src/noc/containers/noc-check-and-submit/noc-check-and-submit.component.ts

```diff
@@ -83,19 +102,30 @@ export class NocCheckAndSubmitComponent implements OnInit, OnDestroy {
   }
 
   public verifyAndSubmitNoC(): void {
-    if (this.affirmationAgreed) {
+    if (this.affirmationAgreed && this.notifyEveryParty) {
```

Comment:
Not a blocker but this series of conditionals includes a fair bit of duplication and is also not the easiest to follow. It could be improved with:

```
let affirmationError;
if (this.affirmationAgreed) {
  if (this.notifyEveryParty) {
    const nocEvent: NocEvent = { case_id: this.caseRefernce, answers: this.nocAnswers };
    this.store.dispatch(new fromFeature.SubmitNoc(nocEvent));
  } else {
    affirmationError = { AFFIRMATION_NOTIFY_EVERY_PARTY_ERROR };
  }
} else {
  if (this.notifyEveryParty) {
    affirmationError = { AFFIRMATION_DEFAULT_DISAGREE_ERROR };
  } else {
    affirmationError = { AFFIRMATION_DEFAULT_DISAGREE_ERROR, AFFIRMATION_NOTIFY_EVERY_PARTY_ERROR };
  }
}
if (affirmationError) {
  this.store.dispatch(new fromFeature.SetAffirmationError(affirmationError));
}
```

Reply:
agreed the duplication which I have amended, but I felt the 2nd level nested if/else block actually reduce the code readability which I would like to avoid, it's also not recommented to use nested if/else block if it has other ways.

Reply:
Agree to disagree. :)